apps/aeoracle/src/aeo_state_tree.erl

```diff
@@ -30,32 +30,45 @@
 %%% Types
 %%%===================================================================
 
--type mkey() :: aeo_oracles:id() | aeo_interaction:id().
--type mvalue() :: aeo_oracles:serialized() | aeo_interaction:serialized().
--type mtree() :: aeu_mtrees:tree(mkey(), mvalue()).
+%% The main tree contains oracles - then there is a set of trees (one per
+%% oracle) that contains the interactions... The 'oracle' field only contain
+%% the root hash of the interaction tree.
+%%
+%% We also keep a cache of objects to expire.
+
+-type otree() :: aeu_mtrees:tree(aeo_oracles:id(), aeo_oracles:serialized()).
+-type itree() :: aeu_mtrees:tree(aeo_interaction:id(), aeo_interaction:serialized()).
 -type interaction() :: aeo_interaction:interaction().
 -type oracle() :: aeo_oracles:oracle().
--type cache() :: gb_sets:set({integer(), binary()}).
+-type itrees() :: gb_tree:tree(aeo_oracles:id(), itree()).
```

Comment:
btw it is gb_trees (with s)

Reply:
One could imagine that Dialyzer can warn about that... Good catch.